core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/operators/aggregate/FinalAggregateOpExec.scala

```diff
@@ -31,33 +34,48 @@ class FinalAggregateOpExec[Partial <: AnyRef](
       pauseManager: PauseManager,
       asyncRPCClient: AsyncRPCClient
   ): Iterator[Tuple] = {
+    if (aggFuncs.isEmpty) {
+      throw new UnsupportedOperationException("Aggregation Functions Cannot be Empty")
+    }
     tuple match {
       case Left(t) =>
-        val groupBySchema = if (aggFunc == null) null else aggFunc.groupByFunc(t.getSchema)
+        val groupBySchema = if (aggFuncs.isEmpty) null else aggFuncs(0).groupByFunc(t.getSchema)
```

Comment:
we don't need to add code check for the empty cases anymore, just assume it's not empty, since we already throw an exception